Replace strip_www with optional redirect to www/non-www

fullyint opened the pull request:
Discussion in #312.

This `www_redirect` is on a per-site basis whereas `strip_www` was for all sites or none.

This includes extra logic to avoid redirecting subdomains, based on comments in #312. However, I wonder if some people would actually want `www.sub.example.com` to redirect to `sub.example.com`. If we can allow the redirect for subdomains, we could drop this portion of the conditional logic:
```
... and host.split('.') | difference('www') | count <= 2
```
(refs: [split](http://stackoverflow.com/a/30517735), [difference](http://docs.ansible.com/ansible/playbooks_filters.html#set-theory-filters), [count/length](http://jinja.pocoo.org/docs/dev/templates/#length))

retlehs commented:
> However, I wonder if some people would actually want www.sub.example.com to redirect to sub.example.com

:+1: ([ignore my previous comment](https://github.com/roots/trellis/issues/312#issuecomment-143929678))

fullyint commented:
Thanks @retlehs. I adjusted to allow redirects for subdomains:
```diff
- {% for host in item.value.site_hosts if item.value.www_redirect | default(true) and host.split('.') | difference('www') | count <= 2 %}
+ {% for host in item.value.site_hosts if item.value.www_redirect | default(true) %}
```

swalkinshaw commented:
Tested and works great :+1: 

Needs a docs update as well.
